src/services/QueryService.ts

```diff
@@ -126,15 +126,18 @@ export class QueryService<T> {
   }
 
   private shouldPullMoreRecords({ count = 0, start = 0, total = 0 }) {
-    const [nextStart, nextCount] = this.getNext(start, count);
+    const { nextStart, nextCount, alreadyFetched } = this.getNext(start, count);
+    if (!this.parameters.showTotalMatched) {
+      return (alreadyFetched < this.parameters.count && count !== 0) ? nextCount : 0;
```

Comment:
```
return (count > 0) ? nextCount : 0;
```

Reply:
I was using alreadyFetched to short circuit when we know we have already gotten as many as the user requested, so we don't have to make a call that returns 0 records every time

Reply:
But you don't need to because you have next count